pallets/common/src/traits/protocol_fee.rs

```diff
@@ -0,0 +1,42 @@
+use codec::{Decode, Encode};
+use frame_support::dispatch::DispatchResult;
+use polymesh_primitives::Signatory;
+#[cfg(feature = "std")]
+use sp_runtime::{Deserialize, Serialize};
+use sp_std::{fmt::Debug, prelude::*};
+
+/// A wrapper for the name of a chargeable operation.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Decode, Encode, Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct OperationName(pub Vec<u8>);
```

Comment:
It looks like we can use a regular `Enum` to represent `OperationName`, instead of a TEXT representation.
It would **reduce the memory** footprint, **eliminate typo-errors**, and **remove transformation** functions.
About storage mapping, we could use `BaseFee: (u8) -> BalanceOf<T>` and `[repr(u8)]` on `OperationName`.

Reply:
I think the enum is a logical solution given that we decided to have protocol fees only in a fixed set of extrinsics. I'll see how to introduce it.